src/libertem/common/buffers.py

```diff
@@ -377,17 +377,23 @@ def get_view_for_dataset(self, dataset):
             raise RuntimeError("Cache is not empty, has to be flushed")
         return self._data
 
+    def _get_slice(self, slice: Slice):
+        real_slice = slice.get()
+        result = self._data[real_slice]
+        # Defend against #1026
+        assert result.shape == tuple(slice.shape) + self.extra_shape
```

Comment:
I think this should not be an `assert`, but an explicit exception `raise`. Maybe a `ValueError`?

Reply:
Since the buffer slicing is used internally and is used in a performance-critcal code path, my thinking was that the `assert` catches issues with upstream use within LiberTEM and then allows to run an optimized build that skips the checks. Or do we have scenarios where an end-user provides the slices?

Reply:
That's true. Maybe you could add a comment for future readers?